src/git/models/pullRequest.ts

```diff
@@ -272,7 +270,7 @@ export function getRepositoryIdentityForPullRequest(
 	if (headRepo) {
 		return {
 			remote: {
-				url: pr.refs?.head?.url,
+				url: pr.repository.url,
```

Comment:
This changes the url ref to the base repository. This would break the model for PRs where the PR is targeting a different repository and (like when the head ref is from someone's fork).

Commands that use this like "open the PR's repository" or "open in worktree" will break in those cases.

Reply:
@axosoft-ramint 
Thank you for catching it.

Question here. `pr.refs?.head?.url` is a link to a branch to to the repo. It fails matching to `uniqueRemoteUrls` in the LaunchpadProvider when I try to `swtichTo` the branch.
So I see 3 ways to fix it:

1. Add `pr.refs.head.repoUrl` on the specific provider level.
2. Fix the matching to `uniqueRemoteUrls` in launchpadProvider
3. Cut it right here: `url: someOperationOnRefUrl(pr.refs?.head?.url)`

I think that the first option is the best because it can respect specifics of each type of integration. What do you think?

Reply:
@sergeibbb `pr.hefs?.head?.url` should be the link to the remote (repository) that the head ref is hosted on, not a link to a branch - at least, with GitHub this is the case. Are you finding that with GitLab, it is linking to a branch? If so, whatever we are using to convert GitLab PRs into our PR model, I think we are plugging the wrong datapoint into that property and should start there.

Reply:
@axosoft-ramint 

GitLab REST API does not provide necessary information about source and target repositories. So, I've fixed it by switching to GraphQL.  (26727ffce4a3546a1b618d38c7c15e4d615d2a4f) 

However I haven't found a good way to search MRs via GraphQL. So I've solved it with 2 requests. I still use REST to search and then I use PRs' IDs to retrieve the rest of the fields from GraphQL. (3cb31d915a128b051a77e026e7d43791f5fbe0a4) 

cc @nzaytsev